packages/windows/src/RPA/Windows/keywords/window.py

```diff
@@ -153,17 +158,17 @@ def foreground_window(
         """
         if locator:
             return self.control_window(locator, foreground=True)
-        if not self.ctx.window.item:
+        window = self.window
+        if window is None:
             raise WindowControlError("There is no active window")
-        utils.call_attribute_if_available(self.ctx.window.item, "SetFocus")
-        utils.call_attribute_if_available(self.ctx.window.item, "SetActive")
-        self.ctx.window.item.MoveCursorToMyCenter(simulateMove=self.ctx.simulate_move)
-        return self.ctx.window
+
+        utils.call_attribute_if_available(window.item, "SetFocus")
```

Comment:
Nice wrapper for safety 👍

Reply:
Thx, although you might get tricked by the diff as it wasn't created/edited by me.
The refactor was: `self.ctx.window` -> `self.window`, so it gets through property wrapper's checks.